src/oc_erchef/apps/oc_chef_wm/src/chef_wm_actions_queue_monitoring.erl

```diff
@@ -0,0 +1,473 @@
+    {reply, ok, State#queue_monitor_state{timer = undefined}};
+handle_call(Request, _From, State) ->
+    lager:debug("Unknown request: ~p", [Request]),
+    {reply, ignored, State}.
+
+
+%%-------------------------------------------------------------
+%% CASTS
+handle_cast(message_dropped, #queue_monitor_state{total_dropped = TotalDropped,
+                                    dropped_since_last_check = Dropped} = State) ->
+    {noreply, State#queue_monitor_state{total_dropped = TotalDropped + 1,
+                          dropped_since_last_check = Dropped + 1}};
+handle_cast(Msg, State) ->
+    lager:debug("Unknown cast: ~p", [Msg]),
+    {noreply, State}.
+
+
+%%-------------------------------------------------------------
+%% INFO
+handle_info(reset_dropped_since_last_check, State) ->
+    {noreply,State#queue_monitor_state{dropped_since_last_check = 0}};
+handle_info({'EXIT', From, Reason}, #queue_monitor_state{worker_process = WorkerPid,
+                                           sync_response_process = SyncPid} = State) ->
+    % Check to see if the EXIT came from our worker
+    case From == WorkerPid of
+        true ->
+              % check if we need to reply to a sync_check_current_state() call
+              case SyncPid of
+                  undefined -> ok;
+                  Pid -> gen_server:reply(Pid, ok)
+              end,
+              % clear the worker_process, allowing for future workers to start
+              {noreply, State#queue_monitor_state{worker_process = undefined,
+                                    sync_response_process = undefined}};
+        false ->
+            lager:warning("Unknown process exit detected in Queue Monitor ~p ~p", [From, Reason]),
+          {noreply, State}
+    end;
+handle_info({MaxLength, N, AtCap}, State) ->
+    % a successful check of the max queue length and current queue length
+    % just update the state and carry on
+    {noreply,State#queue_monitor_state{
+               max_length = MaxLength,
+               last_recorded_length = N,
+               queue_at_capacity = AtCap,
+               dropped_since_last_check = 0
+              }};
+% guard against starting more than one worker process via the
+% match to worker_process = undefined.
+handle_info(status_ping, #queue_monitor_state{worker_process = undefined,
+                                dropped_since_last_check = Dropped} = State) ->
+    ParentPid = self(),
+    Pid = spawn_link(fun () -> check_current_queue_state(ParentPid, Dropped) end),
+    {noreply, State#queue_monitor_state{worker_process=Pid}};
+handle_info(status_ping, State) ->
+    lager:info("Queue monitor check still running, skipping next check"),
+    {noreply, State};
+handle_info(Info, State) ->
+    lager:debug("Unknown info: ~p", [Info]),
+    {noreply, State}.
+
+terminate(_Reason, #queue_monitor_state{timer=undefined}) ->
+    ok;
+terminate(_Reason, #queue_monitor_state{timer=Timer}) ->
+    {ok, cancel} = timer:cancel(Timer),
```

Comment:
Do we want to crash it if timer cancel failed? We can probably safely ignore the result and save some potentially misleading crash log spew.